packages/components/notification/src/notify.ts

```diff
@@ -29,84 +29,86 @@ const notifications: Record<
 const GAP_SIZE = 16
 let seed = 1
 
-const notify: NotifyFn & Partial<Notify> = function (options = {}) {
-  if (!isClient) return { close: () => undefined }
+const notify: NotifyFn & Partial<Notify> & { _context: AppContext | null } =
+  function (options = {}, context: AppContext | null = null) {
+    if (!isClient) return { close: () => undefined }
 
-  if (typeof options === 'string' || isVNode(options)) {
-    options = { message: options }
-  }
+    if (typeof options === 'string' || isVNode(options)) {
+      options = { message: options }
+    }
 
-  const position = options.position || 'top-right'
-
-  let verticalOffset = options.offset || 0
-  notifications[position].forEach(({ vm }) => {
-    verticalOffset += (vm.el?.offsetHeight || 0) + GAP_SIZE
-  })
-  verticalOffset += GAP_SIZE
-
-  const { nextZIndex } = useZIndex()
-
-  const id = `notification_${seed++}`
-  const userOnClose = options.onClose
-  const props: Partial<NotificationProps> = {
-    // default options end
-    zIndex: nextZIndex(),
-    offset: verticalOffset,
-    ...options,
-    id,
-    onClose: () => {
-      close(id, position, userOnClose)
-    },
-  }
+    const position = options.position || 'top-right'
 
-  let appendTo: HTMLElement | null = document.body
-  if (isElement(options.appendTo)) {
-    appendTo = options.appendTo
-  } else if (isString(options.appendTo)) {
-    appendTo = document.querySelector(options.appendTo)
-  }
+    let verticalOffset = options.offset || 0
+    notifications[position].forEach(({ vm }) => {
+      verticalOffset += (vm.el?.offsetHeight || 0) + GAP_SIZE
+    })
+    verticalOffset += GAP_SIZE
 
-  // should fallback to default value with a warning
-  if (!isElement(appendTo)) {
-    debugWarn(
-      'ElNotification',
-      'the appendTo option is not an HTMLElement. Falling back to document.body.'
-    )
-    appendTo = document.body
-  }
+    const { nextZIndex } = useZIndex()
 
-  const container = document.createElement('div')
+    const id = `notification_${seed++}`
+    const userOnClose = options.onClose
+    const props: Partial<NotificationProps> = {
+      // default options end
+      zIndex: nextZIndex(),
+      offset: verticalOffset,
+      ...options,
+      id,
+      onClose: () => {
+        close(id, position, userOnClose)
+      },
+    }
 
-  const vm = createVNode(
-    NotificationConstructor,
-    props,
-    isVNode(props.message)
-      ? {
-          default: () => props.message,
-        }
-      : null
-  )
+    let appendTo: HTMLElement | null = document.body
```

Comment:
`document.body ` SSR compatible ?

Reply:
Note that at line#34 we have a judgement for early termination

```ts
if (!isClient) return {}
```